src/snowflake/snowpark/_internal/compiler/cte_utils.py

```diff
@@ -48,6 +57,10 @@ def traverse(root: "TreeNode") -> None:
             next_level = []
             for node in current_level:
                 id_count_map[node.encoded_node_id_with_query] += 1
+                if propagate_complexity_hist:
+                    id_complexity_map[node.encoded_node_id_with_query].append(
+                        get_complexity_score(node)
+                    )
```

Comment:
shouldn't two nodes with same `encoded_node_id_with_query` have the same complexity? we are simply going to make a list of `[complexity] * id_count_map[node_id]`

Reply:
so i was recording the complexity for each node because theoritically they can be different node end with same query. However, if they are the same query, theoretically i would in general expect the complexity to be same or close, since that information is mainly used for estimation, i think we can just use the repetition_count to make things simpiler